nack/generator_interceptor.go

```diff
@@ -0,0 +1,181 @@
+package nack
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/pion/interceptor"
+	"github.com/pion/logging"
+	"github.com/pion/rtcp"
+	"github.com/pion/rtp"
+)
+
+// GeneratorInterceptor interceptor generates nack feedback messages.
+type GeneratorInterceptor struct {
+	interceptor.NoOp
+	size        uint16
+	skipLastN   uint16
+	interval    time.Duration
+	receiveLogs *sync.Map
```

Comment:
Official document says: https://godoc.org/sync#Map
> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content. 

I think it's better to use strict type for this.

Reply:
It also states: 

> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times

which is the exact use case here